CRM/Pledge/BAO/Pledge.php

```diff
@@ -289,9 +289,9 @@ public static function deletePledge($id) {
    * @param string $startDate
    * @param string $endDate
    *
-   * @return array|null
+   * @return array
    */
-  public static function getTotalAmountAndCount($status = NULL, $startDate = NULL, $endDate = NULL) {
+  public static function getTotalAmountAndCount($status = NULL, $startDate = NULL, $endDate = NULL): array {
```

Comment:
still seeing notices like

>Deprecated function: substr(): Passing null to parameter #1 ($string) of type string is deprecated in CRM_Pledge_BAO_Pledge::getTotalAmountAndCount() (line 327 of /Applications/MAMP/htdocs/d10/vendor/civicrm/civicrm-core/CRM/Pledge/BAO/Pledge.php).
Deprecated function: substr(): Passing null to parameter #1 ($string) of type string is deprecated in CRM_Pledge_BAO_Pledge::getTotalAmountAndCount() (line 328 of /Applications/MAMP/htdocs/d10/vendor/civicrm/civicrm-core/CRM/Pledge/BAO/Pledge.php).

perhaps `$startDate` & `$endDate` should be initialized to empty string instead of NULL?

Reply:
@jitendrapurohit I added handling close to where the string function is used